File: src/main/java/spring/webmvc/presentation/dto/request/MemberUpdateRequest.java
package spring.webmvc.presentation.dto.request;

import java.time.LocalDate;

import jakarta.validation.constraints.Pattern;

public record MemberUpdateRequest(
	String password,
	String name,
	@Pattern(regexp = "^010-\\d{3,4}-\\d{4}$")
	String phone,
	LocalDate birthDate
) {
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/request/MemberUpdateRequest.kt
package spring.web.kotlin.presentation.dto.request

import jakarta.validation.constraints.Pattern
import java.time.LocalDate


data class MemberUpdateRequest(
    val password: String?,
    val name: String?,
    @field:Pattern(regexp = "^010-\\d{3,4}-\\d{4}$")
    val phone: String?,
    val birthDate: LocalDate?,
)
